### PR TITLE
combined project

### PR DESCRIPTION
**Pull request by hillyleopard133**
edited student class and student test class and copied module classes

### CODE REVIEW

**Comment by hillyleopard133** [comment]
i didnt delete your original files, just combined them into my folder, you might want to remove yours then idk

**Comment by evanhearne** [comment]
> i didnt delete your original files, just combined them into my folder, you might want to remove yours then idk

Will take a look shortly. Looks good from the file changes, just need to check the JUnit tests from my end to make sure they work. :) Btw probably best next time to add changes to a branch that's not your main.

File: SoftwareEngineering/src/Student.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    private Module[] modules;
+
+    // Constructor
+
+    public Student(String name, int studentID, String course) {
+        this.name = name;
+        this.studentID = studentID;
+        this.course = course;
+    }
+
```
Comment (evanhearne) [inline SoftwareEngineering/src/Student.java]: A blank constructor needs to be added for all JUnit tests to pass.

File: SoftwareEngineering/test/StudentTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        student3 = new Student("Max", 66643891, "Medicine");
+        student4 = new Student( "Bob", 39084372, "Sociology");
+
+        club1 = new Club("Chess", "Slay the opponents king");
+        club2 = new Club("Rowing", "Join us as we travel down the rivers");
+        club3 = new Club("Handball", "Handball is a sport");
+        club4 = new Club("Drawing", "Drawing supplies are not provided");
+        
+        
+        student = new Student("John", 25, "Bsc Computing);
```
Comment (evanhearne) [inline SoftwareEngineering/test/StudentTest.java]: `"Bsc Computing` needs to be changed to `"Bsc Computing"`

**Review by evanhearne — CHANGES_REQUESTED** [review]
@hillyleopard133 All JUnit tests pass when the changes requested are made. Also, if you could move the src and test files to /src/ and /test/ respectively then our work here is done. Also, if you could do this through your main branch (just this once haha) and push the changes here, then I can go ahead and review changes and hopefully approve the PR.

File: src/Student.java [hunk 3]
```diff
@@ -1,104 +1,152 @@
...
-            Collections.sort(grades);
-            Collections.reverse(grades);
-            moduleL.setGrades(grades.stream().mapToInt(i->i).toArray());
-        }
-        modules = moduleList.toArray(new Module[moduleList.size()]);
-        return getAllGrades();
-    }
-}
+import java.util.ArrayList;
+
```
Comment (evanhearne) [inline src/Student.java:2]: Insert `import java.util.Collections;` here

**Review by evanhearne — APPROVED** [review]
All JUnit tests run, but Collections needs to be imported in order to run some of the tests. Overall, pretty good. Feel free to merge. We can just import on the day, not really a big issue (some may say).

**Pull request merged**